src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -447,13 +449,19 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
         logger.print("[JaCoCo plugin] Saving matched execfiles: ");
         dir.addExecFiles(matchedExecFiles);
         logger.print(" " + Util.join(matchedExecFiles," "));
-        FilePath[] matchedClassDirs = resolveDirPaths(filePath, taskListener, classPattern);
+        FilePath[] matchedClassDirs = {};
+        if (run instanceof AbstractBuild) {
+            matchedClassDirs = resolveDirPaths((AbstractBuild) run, filePath, taskListener, classPattern);
+        }
```

Comment:
Not sure about the Jenkins internas here, but shouldn't we keep the previous behaviour if run is not an AbstractBuild?